public/js/invertedIndex.js

```diff
@@ -0,0 +1,155 @@
+ * @class
+ */
+class InvertedIndex {
+
+  /**
+   * class constructor
+   * @constructor
+   */
+  constructor() {
+    this.unique = [];
+    this.tableObj = {};
+  }
+
+  /**
+   * returns the object containing all the uploadef files
+   * @return {Object} Index of all the files
+   */
+  searchAllFiles() {
+    return this.tableObj;
+  }
+
+  /**
+   * A method that tokenizes the string that is passed through it
+   * @param  {string} data the string that is gotten from the "text" key in the JSON object
+   * @return {Array} containing String.
+   */
+  tokenize(data) {
+    this.terms = data.replace(/[^\w\s]/gi, ' ')
+      .match(/\w+/g);
+    return this.terms;
+  }
+
+  /**
+   * A method that filters the array passed into it for unique words
+   * @param {String} data Array of strings
+   * @returns {String} Returns array of unique words
+   */
+  uniqueWords(data) {
+    this.unique = [...new Set(data)];
+    return this.unique;
+  }
+
+  /**
+   * A method that tokenizes an object value and gets the unique terms in the object values
+   * @param  {Object} fileJsonObject the uploaded JSON file object
+   * @return {Array}  of the unique terms in the fileJsonObject
+   */
+  getTextFromJsonObj(fileJsonObject) {
+    let newText = ' ';
+    Object.keys(fileJsonObject)
+      .forEach((key) => {
+        const obj = fileJsonObject[key];
+        newText += ' ';
+        newText += obj.text;
+      });
+    const uniqueTerms = this.uniqueWords(this.tokenize(newText))
+      .map(x => x.toLowerCase());
+    return uniqueTerms;
+  }
+
+  /**
+   * A method to validate a JSON file.
+   * it check if the format of the .json file contains
+   * strictly keys of ["title","text"]
+   * @param {Object} parsedFile object
+   * @returns {Object} containing boolean and a String.
+   */
+  validateFile(parsedFile) {
+    this.isValid = {
+      valid: false,
+      message: 'This is an Invalid JSON File',
+    };
+    Object.keys(parsedFile)
+      .forEach((key) => {
+        const obj = parsedFile[key];
+        const validFormat = ['title', 'text'];
+        const parsedFileFormat = Object.keys(obj);
+        const arr = Object.keys(parsedFileFormat)
+          .map(objKeys => parsedFileFormat[objKeys]);
+        if (validFormat.toString() === arr.toString()) {
+          this.isValid = {
+            valid: true,
+            message: 'This JSON Format is correct',
+          };
+        } else {
+          this.isValid = {
+            valid: false,
+            message: 'This JSON Format is Incorrect',
+          };
+        }
+      });
+    return this.isValid;
+  }
+
+  /**
+   * A method that create the index of the selected-file
+   * @param  {Object} fileJsonObject the uploaded file object
+   * @param  {Array} uniqueTerms    the unique terms in the JSON file
+   * @param  {String} fileName    The file name
+   * @return {Object} containing the index of the selected-file
+   */
+  createIndex(fileJsonObject, uniqueTerms, fileName) {
```

Comment:
You should not make use of key words when defining variables. renmae the 'fileJsonObject' argument